### PR TITLE
Send Tx All Assets Selected Fix

### PR DESCRIPTION
**Pull request by banklesss**
Related to [YOMO-504](https://emurgo.atlassian.net/browse/YOMO-504)

**All Asset selected**

https://user-images.githubusercontent.com/105349292/234328711-be122894-9cb0-4778-b5e1-4126577e9c1f.mov

**Empty Wallet**

https://user-images.githubusercontent.com/105349292/234328760-855875d2-8b6d-4e90-9f22-83e09477eadf.mov

[YOMO-504]: https://emurgo.atlassian.net/browse/YOMO-504?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[YOMO-460]: https://emurgo.atlassian.net/browse/YOMO-460?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/features/Send/useCases/ListAmountsToSend/AddToken/SelectTokenFromListScreen.tsx [hunk 1]
```diff
@@ -216,23 +219,26 @@ const SelectableAssetItem = ({tokenInfo, disabled, wallet}: SelectableAssetItemP
   )
 }
 
-const ListEmptyComponent = ({fungibilityFilter}: {fungibilityFilter: FungibilityFilter}) => {
+const ListEmptyComponent = ({
```
Comment (banklesss) [inline src/features/Send/useCases/ListAmountsToSend/AddToken/SelectTokenFromListScreen.tsx:222]: Simplification. Nfts out.

File: src/features/Send/useCases/ListAmountsToSend/AddToken/SelectTokenFromListScreen.tsx [hunk 2]
```diff
@@ -216,23 +219,26 @@ const SelectableAssetItem = ({tokenInfo, disabled, wallet}: SelectableAssetItemP
...
+  const amountsSelected = Object.keys(targets[selectedTargetIndex].entry.amounts)
+  const allAssetsSelected = getAllAssetsSelected(amountsSelected, allTokenInfos)
   const strings = useStrings()
-
   const isWalletEmpty = useIsWalletEmpty(wallet)
 
   if (isSearching && assetSearchTerm.length > 0 && filteredTokenInfos.length === 0) return <EmptySearchResult />
 
-  if (fungibilityFilter === 'ft' && isWalletEmpty && !isSearching)
+  if ((isWalletEmpty || allAssetsSelected) && !isSearching)
```
Comment (banklesss) [inline src/features/Send/useCases/ListAmountsToSend/AddToken/SelectTokenFromListScreen.tsx:239]: fix

File: src/features/Send/useCases/ListAmountsToSend/AddToken/SelectTokenFromListScreen.tsx [hunk 3]
```diff
@@ -314,26 +320,26 @@ const Counter = ({fungibilityFilter, counter}: {fungibilityFilter: FungibilityFi
...
-  })
-
   /*
    * to show the empty list component:
    *    - filteredTokenInfos has primary token when the search term and the wallet are empty and the ft or all tab are selected
    *    - "ft" tab has to have primary token hidden when wallet is empty and to show the empty list component
    *    - "all" tab has to display the primary token and not to show the empty list component
    */
-  if (fungibilityFilter === 'ft' && isWalletEmpty && assetSearchTerm.length === 0 && !isSearching) return []
+  if (fungibilityFilter === 'ft' && isWalletEmpty && !isSearching) return []
```
Comment (banklesss) [inline src/features/Send/useCases/ListAmountsToSend/AddToken/SelectTokenFromListScreen.tsx:342]: it was unnecessary

File: src/features/Send/useCases/ListAmountsToSend/AddToken/SelectTokenFromListScreen.tsx [hunk 4]
```diff
@@ -111,8 +111,10 @@ const NftList = ({canAddAmount}: {canAddAmount: boolean}) => {
         onSelect={onSelect}
         readOnly={!canAddAmount}
         isRefreshing={false}
-        withVerticalPadding={nfts.length > 0} // to keep consistency between tabs when the list is not empty
-        ListEmptyComponent={<ListEmptyComponent fungibilityFilter="nft" />}
+        withVerticalPadding={filteredAndSorted.length > 0} // to keep consistency between tabs when the list is not empty
+        ListEmptyComponent={
+          filteredAndSorted.length === 0 ? <NoAssetsYet text={strings.noAssetsAddedYet(strings.nfts(2))} /> : null
+        }
```
Comment (banklesss) [inline src/features/Send/useCases/ListAmountsToSend/AddToken/SelectTokenFromListScreen.tsx:117]: simplification

**Review by michalszorad — APPROVED** [review]

**Comment by stackchain** [comment]
in favor of #2459

**Pull request closed without merging**